### ISSUE TITLE
Singularity: SINGULARITY_CACHEDIR + SINGULARITY_LOCALCACHEDIR useful on ShARC

### ISSUE THREAD

**Issue by willfurnass** [rcgsheffield/sheffield_hpc#797]
If you try to do on-the-fly conversion of a Docker image to a Singularity image (e.g. using `singularity run docker://broadinstitute/gatk:4.0.2.0`) then by default image layers are cached in `~/.singularity/docker`.

Similarly, if you try to explicitly convert a Docker image to a Singularity image file as a one-off process (e.g. using `singularity pull docker://broadinstitute/gatk:4.0.2.0`) then again, image layers are cached in `~/.singularity/docker`.

This is problematic in environments such as ShARC with small (10GB) home dirs.  However, there is a workaround, which is to set `SINGULARITY_CACHEDIR` (used by `singularity pull docker://xxx`) and `SINGULARITY_LOCALCACHEDIR` (used by `singularity run docker://xxx`) to directories on filesystems with more generous or no quotas e.g.

```bash
export SINGULARITY_CACHEDIR="/fastdata/${USER}/singularity-cachedir"
export SINGULARITY_LOCALCACHEDIR="${SINGULARITY_CACHEDIR}"
mkdir -p ${SINGULARITY_CACHEDIR}
```

Both of these environment variables are [documented in the Singularity documentation](http://singularity.lbl.gov/build-environment) but we should highlight their importance in ShARC's documentation and make sure we link to the relevant part of the Singularity documentation.

**Comment by willfurnass** [comment]
@gyengen I'd be interested if you find this is still needed/relevant with the currently installed version of Singularity.

**Comment by willfurnass** [comment]
Setting `SINGULARITY_LOCALCACHEDIR` doesn't seem to do anything anymore.  We should still provide advice on setting `SINGULARITY_CACHEDIR` to somewhere with plenty of space though as it's used when executing `singularity run docker://foo:bar` and `singularity pull something.simg docker://foo:bar`

**Comment by twinkarma** [comment]
@willfurnass Issue sorted? shall we close?

**Comment by willfurnass** [comment]
@twinkarma No, I think we still need to document the use of `SINGULARITY_CACHEDIR`
